src/couch_replicator_manager.erl

```diff
@@ -232,14 +238,28 @@ handle_call(Msg, From, State) ->
     {stop, {error, {unexpected_call, Msg}}, State}.
 
 handle_cast({resume_scan, DbName}, State) ->
-    Since = case ets:lookup(?DB_TO_SEQ, DbName) of
-        [] -> 0;
-        [{DbName, EndSeq}] -> EndSeq
+    Pids = State#state.rep_start_pids,
+    NewPids = case lists:keyfind(DbName, 1, Pids) of
+        {DbName, _Pid} ->
+            Entry = case ets:lookup(?DB_TO_SEQ, DbName) of
+                [] ->
+                    {DbName, 0, true};
+                [{DbName, EndSeq, _Rescan}] ->
+                    {DbName, EndSeq, true}
+            end,
+            true = ets:insert(?DB_TO_SEQ, Entry),
+            Pids;
+        false ->
+            Since = case ets:lookup(?DB_TO_SEQ, DbName) of
+                [] -> 0;
+                [{DbName, EndSeq, _Rescan}] -> EndSeq
+            end,
+            true = ets:insert(?DB_TO_SEQ, {DbName, Since, false}),
+            Pid = start_changes_reader(DbName, Since),
+            couch_log:debug("Scanning ~s from update_seq ~p", [DbName, Since]),
+            [{DbName, Pid} | Pids]
     end,
-    ensure_rep_ddoc_exists(DbName),
```

Comment:
Won't this hurt us? Other cases when we do ensure in this is on init() and on db create event. So if ddoc will be removed after creation, we loose him until next restart.